genericjpa/db/src/main/java/org/hibernate/search/annotations/UpdateInfo.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.annotations;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * @author Martin Braun
+ * @hsearch.experimental
+ */
+@Target({FIELD, METHOD, TYPE})
+@Retention(RUNTIME)
+@Repeatable(UpdateInfos.class)
+public @interface UpdateInfo {
```

Comment:
> Yes, we had it before, and as I said back then, I just think this approach is not really what a  user would expect or be willing to do. It's just too much work on their side.

I don't see why having a mechanism that covers all use cases is bad as a starting point. You have to remember that this mechanism is intended to be used in the generic version as well. After the general mechanism is merged, I am planning on doing a less "hands-on" approach for the users.

> Also it's not clear to me what benefit these annotations provide at all: Their info can be retrieved from the model itself: table name, id type etc. Why asking the user to express this info a second time?

Again, this is intended to be used by both the normal Hibernate Search and the generic version (which I plan on merging after this backend). The normal JPA annotations don't provide enough information so I decided on having everything that is Update mechanism related in the same spot to not confuse people.